exporter/snowflakeexporter/internal/utility/utility.go

```diff
@@ -0,0 +1,96 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package utility provides utility functions for the snowflakeexporter package to consolidate code
+package utility // "github.com/observiq/bindplane-agent/exporter/snowflakeexporter/internal/utility"
+
+import (
+	"database/sql/driver"
+	"encoding/hex"
+	"encoding/json"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+// Array implements driver.Valuer interface allowing arrays to be sent to Snowflake
+type Array []any
+
+// Value marshals the underlying array and then casts to a string value so it can be viewed in Snowflake
+func (a Array) Value() (driver.Value, error) {
+	if a == nil {
+		return nil, nil
+	}
+	bytes, err := json.Marshal(a)
+	if err != nil {
+		return nil, err
+	}
+	return string(bytes), nil
+}
+
+// ConvertAttributesToString converts the pcommon.Map into a JSON string representation
+// this is due to a bug/lacking feature with the snowflake driver that prevents maps from being inserted into VARIANT & OBJECT columns
+// github issue: https://github.com/snowflakedb/gosnowflake/issues/217
+func ConvertAttributesToString(m pcommon.Map, logger *zap.Logger) string {
+	bytes, err := json.Marshal(m.AsRaw())
+	if err != nil {
+		logger.Warn("failed to marshal attribute map", zap.Error(err), zap.Any("map", m))
+		return ""
+	}
```

Comment:
Do we know if it's even possible for json.Marshal to fail here? Like is it possible for a pcommon.Map to include a value that would cause it to fail?

Reply:
Likely it won't fail but it's always good to check the error just in case. If it does happen we'll want to know rather than encountering a panic.

Reply:
Sounds good!